File: InternetBulletin.Shared/DTOs/UserProfileDTO.cs
// *********************************************************************************
//	<copyright file="UserProfileDto.cs" company="Personal">
//		Copyright (c) 2025 Personal
//	</copyright>
// <summary>The User Profile DTO.</summary>
// *********************************************************************************

namespace InternetBulletin.Shared.DTOs
{
	/// <summary>
	/// The User Profile DTO.
	/// </summary>
	public class UserProfileDto
	{
		/// <summary>
		/// Gets or sets the user identifier.
		/// </summary>
		/// <value>
		/// The user identifier.
		/// </value>
		public string UserName { get; set; } = string.Empty;

		/// <summary>
		/// Gets or sets the user posts.
		/// </summary>
		/// <value>
		/// The user posts.
		/// </value>
		public List<UserPostsDto> UserPosts { get; set; } = [];
	}
	
}

File: UI/web/app/Models/UserProfileDto.ts
/**
 * UserProfileDtoModel class to represent the user profile dto model.
 */
class UserProfileDtoModel {
    /**
     * Gets or sets the user identifier.
     */
    userName: string = "";

    /**
     * Gets or sets the user posts.
     */
    userPosts: Array<Object> = [];

    /**
     * Creates an instance of UserProfileDtoModel
     * @param UserName The user identifier.
     * @param UserPosts The user posts.
     */
    constructor(UserName: string, UserPosts: Array<Object>) {
        this.userName = UserName;
        this.userPosts = UserPosts;
    }
}

export default UserProfileDtoModel;
